eslint-plugin-kuc-v1/validator-in-update.js

```diff
@@ -0,0 +1,48 @@
+module.exports = {
+  create: function(context) {
+    return {
+      ClassDeclaration: function(node) {
+        const superClass = node.superClass.name;
+        if (superClass !== "KucBase") return;
+
+        const nodeBody = node.body;
+        if (nodeBody.type !== "ClassBody") return;
+
+        const sourceCode = context.getSourceCode();
+        const baseValidatorRegex = new RegExp('\\./base/validator"');
+        if (!baseValidatorRegex.test(sourceCode.getText(node.parent.parent))) return ;
+
+        const pattern = "validate[a-zA-z]+\\(";
+        const regex = new RegExp(pattern);
+        if (!regex.test(sourceCode.getText())) return;
+
+        nodeBody.body.forEach(body => {
+          if (
+            body.type === "MethodDefinition" &&
+            body.key.type === "Identifier" &&
+            !body.computed
+          ) {
+            const functionSourceCode = sourceCode.getText(body.value);
+            if (
+              (body.key.name === "update" && !regex.test(functionSourceCode)) ||
+              (body.kind !== "constructor" &&
+                body.key.name !== "update" &&
+                regex.test(functionSourceCode))
+            ) {
```

Comment:
Is this rule except for example this logic?

``` 
 constructor(props?: ButtonProps) {
    super();
    const validProps = validateProps(props);
    Object.assign(this, validProps);
  }
```

Reply:
Yes, it is.